### PR TITLE
knative: fix knative internal/external lb assumption

### PR DESCRIPTION
**Pull request by scootweiss**
this change adds all knative services to the `internal` proxy. this is based on the implicit Knative assumption that `External` ingresses will be available on the internal proxy
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/2336

### CODE REVIEW

**Comment by scootweiss** [comment]
/kick
gateway test flake:

```
Step #11 - "test": • Failure [24.031 seconds]
Step #11 - "test": Gateway in memory traffic [It] should work with no ssl and cleans up the envoy config when the virtual service is deleted 
Step #11 - "test": /workspace/gloo/test/e2e/gateway_test.go:210
Step #11 - "test": 
Step #11 - "test":   Timed out after 20.000s.
Step #11 - "test":   Expected
Step #11 - "test":       <int>: 200
Step #11 - "test":   to equal
Step #11 - "test":       <int>: 503
Step #11 - "test": 
Step #11 - "test":   /workspace/gloo/test/e2e/gateway_test.go:245
Step #11 - "test": 
Step #11 - "test":   Full Stack Trace
Step #11 - "test":   github.com/solo-io/gloo/test/e2e_test.glob..func10.1.6.4()
Step #11 - "test":   	/workspace/gloo/test/e2e/gateway_test.go:245 +0xff3
Step #11 - "test":   github.com/onsi/ginkgo/internal/leafnodes.(*runner).runSync(0xc00031db00, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
Step #11 - "test":   	/go/pkg/mod/github.com/onsi/ginkgo@v1.11.0/internal/leafnodes/runner.go:113 +0xe0
Step #11 - "test":   github.com/onsi/ginkgo/internal/leafnodes.(*runner).run(0xc00031db00, 0xc0000f9d68, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
Step #11 - "test":   	/go/pkg/mod/github.com/onsi/ginkgo@v1.11.0/internal/leafnodes/runner.go:64 +0x108
Step #11 - "test":   github.com/onsi/ginkgo/internal/leafnodes.(*ItNode).Run(0xc0008e2700, 0x4093c80, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
Step #11 - "test":   	/go/pkg/mod/github.com/onsi/ginkgo@v1.11.0/internal/leafnodes/it_node.go:26 +0x88
Step #11 - "test":   github.com/onsi/ginkgo/internal/spec.(*Spec).runSam
```

**Comment by marcogschmidt** [comment]
Regression test flake /kick

File: changelog/v1.3.4/fix-knative-internal-gateway.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+changelog:
+  - type: FIX
+    description: Ensure Knative Services are exposed on the internal Knative proxy.
```
Comment (marcogschmidt) [inline changelog/v1.3.4/fix-knative-internal-gateway.yaml]: Can you a bit more detail, e.g. explaining why the change was necessary?

File: projects/knative/pkg/translator/translator_syncer.go [hunk 2]
```diff
@@ -43,6 +46,7 @@ func NewSyncer(externalProxyAddress, internalProxyAddress, writeNamespace string
 		ingressClient:        ingressClient,
 		proxyReconciler:      gloov1.NewProxyReconciler(proxyClient),
 		requireIngressClass:  requireIngressClass,
+		translateProxy:       translateProxy,
```
Comment (marcogschmidt) [inline projects/knative/pkg/translator/translator_syncer.go]: Where is that `translateProxy` variable coming from? I don't see it in the arguments to the function...

**Comment by scootweiss** [comment]
/kick


Step #11 - "test": [AfterEach] in memory
Step #11 - "test":   /workspace/gloo/test/e2e/gateway_test.go:67
Step #11 - "test": 
Step #11 - "test": ------------------------------
Step #11 - "test": • Failure [11.021 seconds]
Step #11 - "test": Gateway in memory traffic [It] should work with no ssl and cleans up the envoy config when the virtual service is deleted 
Step #11 - "test": /workspace/gloo/test/e2e/gateway_test.go:210
Step #11 - "test": 
Step #11 - "test":   Timed out after 10.000s.
Step #11 - "test":   Expected
Step #11 - "test":       <bool>: false
Step #11 - "test":   to be true
Step #11 - "test": 
Step #11 - "test":   /workspace/gloo/test/e2e/gateway_test.go:230
Step #11 - "test":

**Comment by scootweiss** [comment]
/kick

known bug?


Step #19 - "regression-tests-cluster-lock": 
Step #19 - "regression-tests-cluster-lock": Gloo failed to install! Detailed logs available at /builder/home/.gloo/debug.log.
Step #19 - "regression-tests-cluster-lock": Error: installing gloo in ingress mode: could not get apiVersions from Kubernetes: unable to retrieve the complete list of server APIs: custom.metrics.k8s.io/v1beta1: the server is currently unable to handle the request
Step #19 - "regression-tests-cluster-lock": Usage:
Step #19 - "regression-tests-cluster-lock":   glooctl install ingress [flags]
Step #19 - "regression-tests-cluster-lock": 
Step #19 - "regression-tests-cluster-lock": Flags:
Step #19 - "regression-tests-cluster-lock":   -h, --help   help for ingress

**Comment by scootweiss** [comment]
/kick

more gateway e2e failures:


Step #17 - "regression-tests": â€¢ Failure [130.000 seconds]
Step #17 - "regression-tests": Kube2e: gateway with subsets and upstream groups [It] routes to subsets and upstream groups 
Step #17 - "regression-tests": /workspace/gloo/test/kube2e/gateway/gateway_test.go:1094
Step #17 - "regression-tests": 
Step #17 - "regression-tests":   Timed out after 120.000s.
Step #17 - "regression-tests":   Expected
Step #17 - "regression-tests":       <string>: command terminated with exit code 7
Step #17 - "regression-tests":        (exit status 7)
Step #17 - "regression-tests":   to contain substring
Step #17 - "regression-tests":       <string>: red-pod
Step #17 - "regression-tests": 
Step #17 - "regression-tests":   /go/pkg/mod/github.com/onsi/ginkgo@v1.11.0/internal/leafnodes/runner.go:113
Step #17 - "regression-tests": 
Step #17 - "regression-tests":   Full Stack Trace
Step #17 - "regression-tests":   github.com/onsi/ginkgo/internal/leafnodes.(*runner).runSync(0xc0005675c0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
Step #17 - "regression-tests":   	/go/pkg/mod/github.com/onsi/ginkgo@v1.11.0/internal/leafnodes/runner.go:113 +0xe0
Step #17 - "regression-tests":   github.com/onsi/ginkgo/internal/leafnodes.(*runner).run(0xc0005675c0, 0xc0000f9668, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
Step #17 - "regression-tests":   	/go/pkg/mod/github.com/onsi/ginkgo@v1.11.0/internal/leafnodes/runner.go:64 +0x108
Step #17 - "regression-tests":   github.com/onsi/ginkgo/internal/leafnodes.(*ItNode).Run(0xc00000f520, 0x320ac00, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
Step #17 - "regression-tests":   	/go/pkg/mod/github.com/onsi/ginkgo@v1.11.0/internal/leafnodes/it_node.go:26 +0x88

**Comment by ilackarms** [comment]
/kick

rate limit flake:

 "test": 
Step #11 - "test": [2020-02-01 17:18:34.680][8466][info][main] [external/envoy/source/server/server.cc:538] main dispatch loop exited
Step #11 - "test": • Failure [5.013 seconds]
Step #11 - "test": Rate Limit with envoy [It] should rate limit 
Step #11 - "test": /workspace/gloo/test/e2e/ratelimit_test.go:140
Step #11 - "test": 
Step #11 - "test":   Timed out after 5.000s.
Step #11 - "test":   Expected
Step #11 - "test":       <*errors.errorString | 0xc00247cf90>: {
Step #11 - "test":           s: "404 is not TooManyRequests",
Step #11 - "test":       }
Step #11 - "test":   to be nil
Step #11 - "test": 
Step #11 - "test":   /workspace/gloo/test/e2e/ratelimit_test.go:148
Step #11 - "test": 
Step #11 - "test":   Full Stack Trace
Step #11 - "test":   github.com/solo-io/gloo/test/e2e_test.glob..func16.1.4()
Step #11 - "test":   	/workspace/gloo/test/e2e/ratelimit_test.go:148 +0x405
Step #11 - "test":   github.com/onsi/ginkgo/internal
